### PR TITLE
Remove Gen1 pricing from the template

### PR DESCRIPTION
**Pull request by dubiety**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Remove deprecated pricing tier from azure maps create template

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@eriklindeman - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@eriklindeman - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.maps/maps-create/azuredeploy.json [hunk 1]
```diff
@@ -2,43 +2,55 @@
...
             } 
         } 
     }, 
     "resources": [
         {
             "name": "[parameters('accountName')]",
             "type": "Microsoft.Maps/accounts",
-            "apiVersion": "2021-02-01",
-            "location": "global",
+            "apiVersion": "2021-12-01-preview",
```
Comment (stevemunk) [inline quickstarts/microsoft.maps/maps-create/azuredeploy.json:52]: Shouldn't the `apiVersion` be **2023-06-01**? See [here](https://learn.microsoft.com/rest/api/maps-management/accounts)

Comment (dubiety) [reply quickstarts/microsoft.maps/maps-create/azuredeploy.json]: Thanks for pointing this out. Updtaed.

**Comment by azure-quickstarts** [comment]
@eriklindeman - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Review by eriklindeman — APPROVED** [review]

**Pull request merged**